src/NServiceBus.AcceptanceTesting/Support/EndpointBehaviorBuilder.cs

```diff
@@ -23,14 +23,30 @@ public EndpointBehaviorBuilder<TContext> When(Func<IMessageSession, Task> action
         {
             return When(c => true, action);
         }
+        
+        public EndpointBehaviorBuilder<TContext> When(Func<TContext, Task<bool>> condition, Func<IMessageSession, Task> action)
+        {
+            behavior.Whens.Add(new WhenDefinition<TContext>(condition, action));
+
+            return this;
+        }
 
+        // remove in v8
```

Comment:
I did not want to take a dependency on obsoleteex nor did I want to break existing usages in this minor release. So I figured I add now a comment.

Reply:
Perhaps also raise an issue for the removal and assign it to https://github.com/Particular/NServiceBus/milestone/193?

That way we won't forget

Reply:
Or maybe since this is the public API for When definitions we don't want to break tests when the upgrade to v8? So the better decision might be to just remove the WhenDefinition ctor overloads?

Reply:
Raised #5215  and removed the comment for now